.github/workflows/dev.yml

```diff
@@ -5,6 +5,8 @@ name: CI
 on:
   push:
     branches: ["dev"]
+    # Match tags that resemble a version
+    tags: ['v[0-9]*', '[0-9]+.[0-9]+*']
```

Comment:
For the tags on the images what format is going to be expected?

If I understand this right you are currently expecting 2 formats:

1. v
2. v1
3. v2
4. v333333333
5. ect....
OR

1. 1.1
2. 2.2
3. 3.0000000000000000

My suggestion:
The most common image versioning I've seen is `x.x.x` -> `[0-9]+\.[0-9]+\.[0-9]+`:
https://regex101.com/
![image](https://github.com/Sage-Bionetworks/agora-data-tools/assets/17128019/6d807ca4-f0ff-4f6d-b327-709e6d5135d4)

Reply:
We use this same `tags` regex across repos in the team. I think the idea is to be more permissive to make sure a new version of the container is always tagged correctly, though I do agree that standardizing how we do versioning (and therefore being able to be more strict with this regex) might be beneficial overall.

- [Example 1](https://github.com/Sage-Bionetworks-Workflows/py-orca/blob/main/.github/workflows/ci.yml)
- [Example 2](https://github.com/Sage-Bionetworks-Workflows/py-dcqc/blob/main/.github/workflows/ci.yml)

Reply:
> I think the idea is to be more permissive to make sure a new version of the container is always tagged correctly

IMO having the regex be wider is going to lead to future problems. Think of a situation where 2 devs are making changes and are allowed to make this tag version history:

1. v1.0.0
2. v1.1.0
3. 1.1.1
4. v1.1.2
5. 1.1.3

Suppose you are testing a breaking change that was introduced in `1.1.3` so you update NF to use `1.1.2`- But there is a problem since that doesn't exist.

@thomasyu888 Thoughts?

Some examples how we are already diverging between projects:
![image](https://github.com/Sage-Bionetworks/agora-data-tools/assets/17128019/ad5c28a0-e2b2-4298-abc9-f6cfe1638c49)
![image](https://github.com/Sage-Bionetworks/agora-data-tools/assets/17128019/0a16d1f9-7124-4d41-9e79-e29436571332)

Reply:
I agree with @BryanFauble here.  Let's update to `[0-9]+\.[0-9]+\.[0-9]+` to reduce the overall complexity.  It's a bit unfortunate that we have two standards, but 

* agora-data-tools = `[0-9]+\.[0-9]+\.[0-9]+`
* py-orca = `v[0-9]+\.[0-9]+\.[0-9]+`

Reply:
updated in my latest commit